### DISCUSSION TITLE
I want to automate code formatting with php-cs-fixer using Github Actions

### DISCUSSION THREAD

**Discussion by gensui0530** [community/community#24629] [Apps, API and Webhooks]
<h3>
<a name="assumption-1" class="anchor" href="#assumption-1"></a>Assumption.</h3>
<p>We are currently looking to automate code formatting with php-cs-fixer using GithubActions.<br>
The yml file is attached for your reference.<br>
The php-cs-fixer command ( <code>. /vendor/bin/php-cs-fixer fix --diff -vvv --dry-run . || true</code> ) to specify all directories would result in a huge amount of code formatting, which we would like to avoid.</p>
<h3>
<a name="what-we-want-to-do-1-2" class="anchor" href="#what-we-want-to-do-1-2"></a>What we want to do: 1.</h3>
<p>When pushing to Github, determine and code format only files that have been modified locally, not all files.</p>
<h3><a name="h-3" class="anchor" href="#h-3"></a></h3>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/7/9792cd2cdda427d7f8a2348e2081349eb2714d8e.png" data-download-href="/uploads/short-url/lCSCmoakxCFv106eJVCAYgpWI0u.png?dl=1" title="スクリーンショット 2022-04-05 12.21.32" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/59169772/181085739-f3f39115-1fd5-4c89-abe2-b94861310636.png" alt="スクリーンショット 2022-04-05 12.21.32" data-base62-sha1="lCSCmoakxCFv106eJVCAYgpWI0u" width="537" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/7/9792cd2cdda427d7f8a2348e2081349eb2714d8e_2_537x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/7/9792cd2cdda427d7f8a2348e2081349eb2714d8e_2_805x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/7/9792cd2cdda427d7f8a2348e2081349eb2714d8e_2_1074x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/7/9792cd2cdda427d7f8a2348e2081349eb2714d8e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">スクリーンショット 2022-04-05 12.21.32</span><span class="informations">2010×1870 370 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
Question.<br>
Is it possible to use the Github Actions feature to determine which files have been committed and execute specific commands only for those files?
<p>If anyone can help us out, we would appreciate it if you could let us know.</p>

**Comment by gensui0530** [comment]
<p>(post deleted by author)</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="gensui0530" data-post="1" data-topic="243193">
<div class="title">
<div class="quote-controls"></div>
 gensui0530:</div>
<blockquote>
<p>Is it possible to use the Github Actions feature to determine which files have been committed and execute specific commands only for those files?</p>
</blockquote>
</aside>
<p>You can get the event payload from the <code>github.event</code> object or read the same data form <code>github.event_path</code> (see <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#github-context"><code>github</code> context</a>). The structure of the event payload depends on the event that triggered the workflow, in your case <code>push</code> or <code>pull_request</code>. You can find documentation on the event payloads here: <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads" class="inline-onebox">Webhook events and payloads - GitHub Docs</a></p>
<p>For a push you can get the commit before and after, or for a PR the head (or temporary merge) and base commits, and compare them.</p>
<p>Note that by default <code>actions/checkout</code> gets <em>only</em> the commit for which the event was triggered. To compare commits you’ll need to fetch the relevant older ones too, or set <a href="https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches"><code>fetch-depth</code> for <code>actions/checkout</code></a>.</p>

**Comment by gensui0530** [comment]
<blockquote>
<p>Note that by default <code>actions/checkout</code> gets <em>only</em> the commit for which the event was triggered. To compare commits you’ll need to fetch the relevant older ones too, or set <a href="https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches" rel="noopener nofollow ugc"> <code>fetch-depth</code> for <code>actions/checkout</code> </a>.</p>
</blockquote>
<p>Thanks!<br>
I did not know this.</p>
<p>I would like to run the following command only on files that have been changed within the PR.</p>
<p><code> /vendor/bin/php-cs-fixer fix --diff -vvv --dry-run || true</code></p>
<p>github.event to get the event payload from the object or<br>
The same data form can be read github.event_path</p>
<p>Is it possible to achieve this with these?<br>
Also, is it possible for you to tell us exactly how to do this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="gensui0530" data-post="4" data-topic="243193">
<div class="title">
<div class="quote-controls"></div>
 gensui0530:</div>
<blockquote>
<p>Is it possible to achieve this with these?</p>
</blockquote>
</aside>
<p>If you can apply your tool to a list of files, yes. Once you have the two commits you want to compare you can use <a href="https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---name-only"><code>git diff --name-only</code></a> to get a list of files that changed between them.</p>
<p>The most effort will probably be to get the right commit IDs to compare.</p>
<p>For a <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push"><code>push</code> event</a> it’s fairly easy at a glance with the <code>before</code> and <code>after</code> fields, except in case of a force-push (in that case the <code>before</code> commit won’t be part of the branch history any more).</p>
<p>For a <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request"><code>pull_request</code></a> you need to decide what you want to compare: There’s the base branch (probably a given), but do you want to compare to the head branch (that’s supposed to get merged) or the temporary merge commit that represents what the result of the merge would be (which is what the workflow will check out by default).</p>
